### PR TITLE
Load GradleProperties earlier

### LINKED ISSUE #1001: Init scripts for buildSrc should have access to gradle.properties
Currently, when having an init script in the Gradle user home directory which uses system properties from a `gradle.properties` in the Gradle user home dir then these are not available when applying the init script to `buildSrc`. Note that these only happens when using the standalone `gradle` command, when using the wrapper then this works.

### Expected Behavior
`gradle.properties` in the Gradle user home are always available to init scripts.

### Current Behavior
When using the `gradle` executable then system properties defined in `gradle.properties` are not available to init scripts applied to `buildSrc`.

### Context
I ran into this issue when using [remoteHttpCacheInit.gradle](https://github.com/gradle/gradle/blob/5b2d6c617896bce058fd2f306eae7e3e23b1ab9a/gradle/remoteHttpCacheInit.gradle) from my Gradle user home. I built a new Gradle version and tried to use that distribution to build Gradle itself.

### Steps to Reproduce (for bugs)

See the failing integration test [here](https://github.com/gradle/gradle/blob/2a4b973481ed90db4ce13f88fad0b1eaf5bc0df1/subprojects/launcher/src/integTest/groovy/org/gradle/launcher/BuildEnvironmentIntegrationTest.groovy#L227-L246).

### LINKED ISSUE #6825: System properties in gradle.properties are non read from init script when using intellij project import
I have a project which sets a system property in gradle.properties. I have an init script reading that property. When I run my project through CLI I see the expected value. When I import the project through IntelliJ I see a default value.

### Expected Behavior
I would like to see the same value read in both cases. In my case the boolean value `true`

### Current Behavior
In CLI, my value is correct `true` but when IntelliJ import the project it is `false`

### Context
I use this to enable experimental features in my plugin which affect what configurations appear in my project. I read system properties later in my plugin too. In the plugin, I see the right value which is not consistent with the value what init script got. This is causing different inconsistent configuration leading to failure with dependency locking.

### Steps to Reproduce (for bugs)
Example project: https://github.com/chali/system-properties-not-honored
* Assuming that you are in project directory cp init.gradle ~/.gradle/init.d/
* Run `./gradlew help` You will see this as part of the output `Example property is: true`
* Run `idea .` and as part of Build: Sync you will see `Example property is: false`

### Your Environment
Gradle 4.10.2
IntelliJ IDEA 2018.2 (Ultimate Edition)
Build #IU-182.3684.101, built on July 24, 2018

JRE: 1.8.0_152-release-1248-b8 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.13.6

### LINKED ISSUE #13122: "GradleProperties has not been loaded yet" error when building project with ProjectBuilder
### Expected Behavior
Gradle introduced new pipe line `providerFactory.gradleProperty` for accessing gradle properties to support instant execution. This new pipeline should be compatible with building Project using ProjectBuilder in unit test. In other words, Gradle should invoke the gradle property loading mechanism at `projectBuilder.build()`   

### Current Behavior
When applying android gradle plugin to the project and accessing some gradle property with the new API, we get "GradleProperties has not been loaded yet" error.

	at org.gradle.initialization.DefaultGradlePropertiesController$NotLoaded.gradleProperties(DefaultGradlePropertiesController.java:74)
	at org.gradle.initialization.DefaultGradlePropertiesController$SharedGradleProperties.gradleProperties(DefaultGradlePropertiesController.java:59)
	at org.gradle.initialization.DefaultGradlePropertiesController$SharedGradleProperties.find(DefaultGradlePropertiesController.java:50)
	at org.gradle.api.internal.provider.sources.GradlePropertyValueSource.obtain(GradlePropertyValueSource.java:36)
	at org.gradle.api.internal.provider.sources.GradlePropertyValueSource.obtain(GradlePropertyValueSource.java:24)
	at org.gradle.api.internal.provider.DefaultValueSourceProviderFactory$ValueSourceProvider.obtainValueFromSource(DefaultValueSourceProviderFactory.java:248)
	at org.gradle.api.internal.provider.DefaultValueSourceProviderFactory$ValueSourceProvider.lambda$calculateOwnValue$0(DefaultValueSourceProviderFactory.java:225)
	at org.gradle.internal.Try.ofFailable(Try.java:39)
	at org.gradle.api.internal.provider.DefaultValueSourceProviderFactory$ValueSourceProvider.calculateOwnValue(DefaultValueSourceProviderFactory.java:225)
	at org.gradle.api.internal.provider.AbstractMinimalProvider.getOrNull(AbstractMinimalProvider.java:103)
	at com.android.build.gradle.options.ProjectOptions.getValue(ProjectOptions.java:144)
	at com.android.build.gradle.internal.DependencyConfigurator.configureGeneralTransforms(DependencyConfigurator.kt:103)
	at com.android.build.gradle.internal.plugins.BasePlugin.createAndroidTasks(BasePlugin.java:601)
	at com.android.build.gradle.internal.plugins.PluginExtensionsKt.runAfterEvaluate(PluginExtensions.kt:24)
	at com.android.build.gradle.internal.plugins.PluginExtensionsKt.runAfterEvaluate$default(PluginExtensions.kt:23)
	at com.android.build.gradle.internal.tests.ConfigurationExtensionTest.setUp(ConfigurationExtensionTest.kt:235)

### Context
We are migrating how we obtain gradle properties from `project.getExtensions().getExtraProperties()` to `providerFactory.gradleProperty`. This issue blocks us as our unit tests no longer work.

### PR DESCRIPTION
**Pull request by oehme**
Fixes #6825
Fixes #13122
Fixes #1001

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by oehme** [comment]
Reading the code, I wonder whether it is intentional that included builds can mutate the System properties with their own values from gradle.properties?

File: subprojects/launcher/src/integTest/groovy/org/gradle/launcher/BuildEnvironmentIntegrationTest.groovy [hunk 1]
```diff
@@ -227,8 +227,8 @@ task check {
     }
 
     @Issue("https://github.com/gradle/gradle/issues/1001")
+    @ToBeFixedForConfigurationCache(because = "Init scripts don't have access to providers yet")
```
Comment (oehme) [inline subprojects/launcher/src/integTest/groovy/org/gradle/launcher/BuildEnvironmentIntegrationTest.groovy:230]: @eskatos FYI, a missing feature I think

**Comment by oehme** [comment]
@big-guy maybe something for you to look at?

File: subprojects/launcher/src/integTest/groovy/org/gradle/launcher/BuildEnvironmentIntegrationTest.groovy (hunk 1 above)
Comment (eskatos) [reply subprojects/launcher/src/integTest/groovy/org/gradle/launcher/BuildEnvironmentIntegrationTest.groovy:230]: Indeed! I just opened https://github.com/gradle/gradle/issues/16849 for this

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by blindpirate** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test RFN

**Review by eskatos — APPROVED** [review]

**Review by eskatos — CHANGES_REQUESTED** [review]
It looks like #13122 isn't fixed.
`DetermineBaselinesTest` fails if we remove this line:

https://github.com/gradle/gradle/blob/0b3bfefa7ed6729f194efd988ed3ac9d00315978/build-logic/performance-testing/src/test/kotlin/gradlebuild/performance/tasks/DetermineBaselinesTest.kt

**Comment by oehme** [comment]
What's the failure? I removed the same line [here](https://github.com/gradle/gradle/pull/16757/files#diff-63913c829037e06e769021884174b4330b8dd52a7d8a1b8cad89cab5f57c785dL279) and the test passes, so I'd say that issue is fixed.

**Review by eskatos — APPROVED** [review]
I've been tricked :) This test runs against the current wrapper. Once the wrapper updated the test passes with the workaround removed :+1: 

This PR must be followed by a wrapper update and by the removal of the workaround in this test.

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Pull request merged**